conversion-measurement/sites/home/shared.browser-setup.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const MIN_SUPPORTED_BROWSER_VERSION = 105
+const SUPPORTED_USER_AGENTS = ['Google Chrome']
+
+displayBrowserVersionRequirements()
+displayWarningBannerIfSetupIncorrect()
+
+function checkIsBrowserSupported() {
+  if (!navigator.userAgentData) {
+    return false
+  }
+  const supportedAgentsByThisUser = navigator.userAgentData.brands.filter(
+    (item) => SUPPORTED_USER_AGENTS.includes(item.brand)
+  )
+  if (supportedAgentsByThisUser.length === 0) {
+    return false
+  } else {
+    const browserVersion = Number.parseFloat(
+      supportedAgentsByThisUser[0].version
+    )
+    if (browserVersion < MIN_SUPPORTED_BROWSER_VERSION) {
+      return false
+    }
+    return true
+  }
+}
```

Comment:
If we're going to use an array of supported user-agent brands, we should probably also specify the minimum version alongside them instead of assuming that each brand has the same minimum version. Otherwise, the list seems unnecessary.

In other words, I'd do something like:

```js
const MIN_SUPPORTED_BRAND_VERSIONS = {
  'Google Chrome': 105,
};

function isBrowserSupported() {
  return navigator.userAgentData?.brands.some(({brand, version}) => {
    const minVersion = MIN_SUPPORTED_BRAND_VERSIONS[brand];
    return Number.parseFloat(version) >= minVersion;
  });
}
```

Reply:
SGTM, updated the code accordingly:
https://github.com/GoogleChromeLabs/trust-safety-demo/pull/34/files#diff-95956c9070e63fd7b9c9d2b916efbc158993f6e231f4ff628fc72a85b99fca3bR17